content/articles/machine-learning-regression-evaluation-metrics/index.md

```diff
@@ -0,0 +1,167 @@
+author: denis-kuria
+date: 2021-11-25T00:00:00-13:00
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/machine-learning-regression-evaluation-metrics/hero.jpg 
+    alt: Machine Learning Regression Evaluation Metrics Hero image
+---
+In machine learning, a regression model is a type of model that predicts a numeric value. These values can be those of price, fees, score, etc. 
+<!--more-->
+We need to measure the performance of machine learning models in order to determine their reliability. After model fitting, we assess the performance of the model by comparing the predictions the model generates to actual data. 
+
+In this tutorial, we are going to discuss various evaluation metrics used in regression models. Besides, we will discuss the advantages and disadvantages of these regression model metrics. Finally, we will discuss what a good regression model is.
+
+### Prerequisites
+To understand this article, you ought to:
+- Be familiar with Machine learning.
+- Be familiar with regression models.
+
+### Table of contents
+- [Introduction](#introduction)
+- [Mean Absolute Error (*MAE*)](#mean-absolute-error-mae)
+- [Mean Squared Error (*MSE*)](#mean-squared-error-mse)
+- [Root Mean Squared Error(*RMSE*)](#root-mean-squared-error-rmse)
+- [R Squared (*R²*)](#r-squared-r²)
+- [Adjusted R Squared](#adjusted-r-squared)
+- [Root Mean Squared Log Error(*RMSLE*)](#root-mean-squared-log-error)
+- [What is a good regression model](#what-is-a-good-regression-model)
+- [Conclusion](#Conclusion)
+
+### Introduction
+There are numerous regression evaluation metrics. All these metrics aim at showing us the prediction error of our model. An error is defined as the difference between the value the model predicts and the actual value. The lower the error, the better the performance of the model, and the higher the error, the worse the performance of the model.
+
+![Reference](/engineering-education/machine-learning-regression-evaluation-metrics/reference.jpeg)
+
+The diagram above shows what an error is and how it is computed.
+
+### Mean Absolute Error (MAE)
+We obtain *MAE* by calculating the absolute difference between the predictions made by the model and the real values. Absolute means that we do not put into consideration the sign of the difference obtained. *MAE* is calculated using the formula below.
+
+![MAE](/engineering-education/machine-learning-regression-evaluation-metrics/mae.png)
+
+*n* is the total number of data points.
+
+We calculate *MAE* by following the steps below:
+1. Calculate the residual (error) of each data point.
+2. Calculate the absolute value. This is to get rid of the sign.
+3. Calculate the average of all residuals.
+
+If *MAE* is zero, the model predictions are perfect. The higher the *MAE* the worse the predictions are.
+
```

Comment:
Perfect? Really? I don't think model predictions can be perfect. Please use the correct word.